client/src/Containers/QuizArea.jsx

```diff
@@ -1,17 +1,214 @@
+        selectdButtonRef.current = '';
+    }
+    // If any other button is selected again
+    else
+    {
+        colorButton(optionID , '#2196f3' , 'white'); // color the currently selected button
+        colorButton(selectedButton , 'white' , 'black'); // uncolor the previously selected button
+        setSelectedButton(optionID); // set the currently selected button
+        selectdButtonRef.current = optionID;
+      }
+  }
+
+  function checkAnswer()
+  {
+    
+    clearInterval(interval); // [ NOT WORKING ] stop the timer
+
+    showAnswerRef.current = true;
+    setShowAnswer(showAnswerRef.current); // set the state of page to show the answer
+
+    if(selectdButtonRef.current === correctOptionRef.current) // if the option selected by user is correct option
+    {
+        colorButton(correctOptionRef.current , '#00e676' , 'white'); // show the correct option in green
+        setScore( prevscore => {
+            let newscore = timeRef.current / timeGivenforQ * 1000;
+            newscore += prevscore;
+            return newscore;
+        }); // update the score
+    }
+    else // if the option selected by user is incorrect
+    {
+        colorButton(correctOptionRef.current , '#00e676' , 'white'); // show the correct option in green
+        colorButton(selectdButtonRef.current , '#f50057' , 'white'); // show the selected incorrect option in red
+    }
+  }
+
+  
+  useEffect(() => {
+        axios.get(`http://localhost:5000/quiz/${quizid}`)
+        .then(data => {
+            setQuizData(data.data);
+            setIsLoading(false);
+            setIndex(0);
+        })
+        .catch( err => alert(`The following error occured : ${err.message}`));
+
+        // this use effect would run whenever this page loads
+        // it fetches the quiz data from DB and displays it to user
+
+        // return () => {
+        //     clearInterval(interval);
+        // }
+  }, []);
+
+  useEffect(() => {
+
+    // This use effect would run whenever the index value of question changes
+    // It would change the question, its corresponding options on quiz interface
+    // and would also reset the timer
 
+    correctOptionRef.current = '';
+    selectdButtonRef.current = '';
+    setSelectedButton(selectdButtonRef.current);
 
+    
+
+    if(isLoading) // if the quiz data is not fetched yet
+    {
+        timeRef.current = 0;
+        setTimeGivenforQ(1);
+    }
+    else // when the quiz data has been fetched
+    {
+        timeRef.current = quizData.questions[index].time_given; // reset the timer value to the time allotted for the question
+        setTime(timeRef.current); // set the time value
+        setTimeGivenforQ(timeRef.current); // set the original time allotted for the given question
+    }
+
+    if(interval) // [ NOT WORKING ] if timer is running then stop it
+    clearInterval(interval);
+      
+    // -------------------------------Timer------------------------------------------------

```

Comment:
Use this (/*************** Timer ****************/) style of commenting for a block of code.